### ISSUE TITLE
How to add a new species to this?

### ISSUE THREAD

**Issue by tariquesani** [tphakala/birdnet-go-classifiers#2]
Hello, 
Came over here from birdnet-go. Been struggling with exactly what this project solves. Improving detection of some very common species from my region which are being missed by 2.4 model. I would appreciate if you can guide / add a small note on the training pipeline 

Thanks
Tarique

**Comment by tariquesani** [comment]
<html>
<body>
<html><head></head><body><p><b>Again answering my own question. This is what I got from Claude. If you can validate it it would be great!</b>

<hr>
Here's a complete guide to building your own House Sparrow-augmented classifier for BirdNET-Go.</p>

<h2>Building a Custom House Sparrow Classifier for BirdNET-Go</h2>
<p>The approach used in that repo is <strong>"append mode" training</strong> — you train on top of BirdNET v2.4, adding or boosting specific classes without replacing the rest of the model. Here's the full process:</p>
<hr>
<h3>1. Install BirdNET-Analyzer</h3>
<p>This is the tool used to do the actual training:</p>
<pre><code class="language-bash">pip install birdnet-analyzer
</code></pre>
<p>Or clone the repo for GUI access: <code>https://github.com/birdnet-team/BirdNET-Analyzer</code></p>
<hr>
<h3>2. Gather Audio Training Data</h3>
<p>This is the most important step. You need <strong>3-second audio clips</strong> of House Sparrow calls.</p>
<p><strong>Good free sources:</strong></p>
<ul>
<li><strong>Xeno-canto</strong> (xeno-canto.org) — search for <em>Passer domesticus</em>, filter by quality A/B recordings</li>
<li><strong>Macaulay Library</strong> (Cornell)</li>
<li>Your own BirdNET-Go recordings that were already correctly identified</li>
</ul>
<p>The quality of your classifier depends heavily on having diverse and representative samples of both positive and negative matches.</p>
<p><strong>Target collection:</strong></p>
<ul>
<li>Minimum 20–50 clips, ideally 100+</li>
<li>Mix of chirps, calls, alarm notes, flock chatter</li>
<li>Variety of recording conditions (different times of day, backgrounds)</li>
</ul>
<hr>
<h3>3. Organise Your Folder Structure</h3>
<p>BirdNET requires a specific folder structure for training data. Each folder represents a class that the AI model will learn to recognize. Folder names follow the pattern: <code>ScientificName_CommonName</code>.</p>
<pre><code>training_data/
├── Passer domesticus_House Sparrow/
│   ├── clip_001.wav
│   ├── clip_002.flac
│   └── ...
├── Background/
│   ├── ambient_garden_1.wav
│   ├── traffic_noise.wav
│   └── ...
</code></pre>
<p>The most important folder for a well-performing classifier is the Background (or Noise) folder. This Background class is essential — without it, the classifier will struggle to distinguish between actual target sounds and random environmental noise, leading to many false positives.</p>
<p>For the Background folder, use ambient recordings from wherever your BirdNET-Go device is deployed — garden noise, wind, rain, traffic. Aim for at least as many clips as your positive class.</p>
<hr>
<h3>4. Train Using BirdNET-Analyzer</h3>
<p><strong>Via GUI:</strong> Open BirdNET-Analyzer → Train tab → select your training data directory → set output path → configure settings below.</p>
<p><strong>Via CLI:</strong></p>
<pre><code class="language-bash">python -m birdnet_analyzer.train \
  --i "path/to/training_data" \
  --o "path/to/output/HouseSparrow_Classifier" \
  --epochs 100 \
  --batch_size 32 \
  --learning_rate 0.0001 \
  --model_save_mode append \
  --model_format tflite \
  --upsampling_ratio 0.5 \
  --mixup \
  --cache_mode save \
  --cache_file train_cache.npz
</code></pre>
<p><strong>Key settings explained:</strong></p>

Parameter | Value | Why
-- | -- | --
--model_save_mode append | append | Critical — adds House Sparrow on top of BirdNET v2.4 without wiping existing birds
--epochs | 100–200 | More epochs = better fit, but watch for overfitting
--upsampling_ratio | 0.5 | Compensates if your sparrow clips are fewer than background
--mixup | flag | Creates synthetic training samples by linearly combining pairs of samples and their labels, improving generalisation
--cache_mode save | save | Stores extracted embeddings in a cache file which can be loaded in later training runs, saving time when retraining


<p>When using append mode, your custom classes are added as new labels to the existing BirdNET model. If a label already exists in the base model, BirdNET-Go will intelligently handle matching labels as a single species, preventing duplicate detections. Since House Sparrow (<em>Passer domesticus</em>) <strong>is already</strong> in BirdNET v2.4, appending it will boost/refine that existing class rather than create a duplicate — exactly what you want.</p>
<hr>
<h3>5. Output Files</h3>
<p>After training you'll get:</p>
<ul>
<li><code>HouseSparrow_Classifier.tflite</code> — the model</li>
<li><code>HouseSparrow_Classifier_Labels.txt</code> — species list</li>
</ul>
<hr>
<h3>6. Deploy to BirdNET-Go</h3>
<p>Copy both files to your BirdNET-Go models directory (<code>birdnet-go-app/data/models/</code>) then update your <code>config.yaml</code>:</p>
<pre><code class="language-yaml">birdnet:
  modelpath: models/HouseSparrow_Classifier.tflite
  labelpath: models/HouseSparrow_Classifier_Labels.txt
</code></pre>
<p>Restart BirdNET-Go.</p>
<hr>
<h3>Tips for House Sparrow Specifically</h3>
<ul>
<li>House Sparrow has a wide vocal repertoire — include <strong>chirps, cheeps, alarm calls, and colony chatter</strong> to cover all call types</li>
<li>Since you're in Nagpur, Maharashtra, try to find recordings from the Indian subcontinent on Xeno-canto for geographic relevance</li>
<li>Iterative refinement: test your classifier and retrain with adjusted parameters or additional data as needed. Start with ~50 clips, test it for a week, then add more data to address any gaps</li>
<li>If you're getting too many false positives after deployment, add more clips of common local confusable sounds (sparrows can get confused with other small passerines) to your Background folder</li>
</ul>
<p>Would you like help writing a script to batch-download and trim Xeno-canto recordings for <em>Passer domesticus</em> to build your training set?</p></body></html>
</body>
</html>

**Comment by tariquesani** [comment]
And I also found this https://github.com/tphakala/birdnet-go/blob/main/doc/wiki/training-a-custom-classifier.md
